### ISSUE TITLE
Not all contacts/closed groups sync

### ISSUE THREAD

**Issue by chairmantheboring** [oxen-io/session-desktop#1563]
**Describe the bug**
In Session desktop, I have experienced issues where a contact fails to sync multi-device, which may also result in #1564. I have also had a closed group fail to sync between devices, leaving me only able to use the group on my phone. I have only experienced this a total of 3 times and it only comes up occasionally; I'm not able to reproduce it on demand.

**To Reproduce**

0. Clear data on the device (Not sure if this matters, but I always cleared data and then linked the same Session ID on the same device again)
1. Link a new device.
2. Check your group and contacts lists. 

**Other information (please complete the following information):**

* Device: PC
* OS: Arch Linux
* Session Version or Git commit hash: Session Desktop 1.5.2 (Flatpak)

Clearing data and reimporting your Session ID may be a potential workaround to this, but it would be better if it didn't happen at all.

**Comment by KeeJef** [comment]
I've seen this too, but usually after restarting the client it fetches the required messages to be back in sync

**Comment by trizen** [comment]
Could this be the reason?
![sshot](https://user-images.githubusercontent.com/614513/120105418-4e46c900-c148-11eb-8eb7-06ec65e7452b.jpg)

**Comment by KeeJef** [comment]
Yeah I'm getting similar logs, something we are looking into

**Comment by rhubear25** [comment]
I have a poss related issue.
I use 2 devices:  phone/PC.
I was in a closed group previously.
Due to issued I was experiencing inside the group, suggestion was to LEAVE the group, which I did... on both devices.
Later I was added into the group again.  Add worked fine on Android, but its showing up on Win10 as "you have left the group" on the message typing bar.
![SessionClosedGroup-ReAdding](https://user-images.githubusercontent.com/57231107/125073015-e8d0eb00-e0aa-11eb-89e6-79049557f7e0.jpg)
